client/fileutils.c

```diff
@@ -649,6 +649,79 @@ int loadFileDICTIONARY(const char *preferredName, void *data, size_t *datalen, u
     return retval;
 }
 
+int loadFileDICTIONARY_safe(const char *preferredName, uint8_t **data, uint8_t keylen, uint16_t *keycnt) {
+
+    int block_size = 512;
+    int allocation_size = block_size;
+    size_t counter = 0;
+    int retval = PM3_SUCCESS;
+    char *path;
+    if (searchFile(&path, DICTIONARIES_SUBDIR, preferredName, ".dic") != PM3_SUCCESS)
+        return PM3_EFILE;
+
+    // t5577 == 4bytes
+    // mifare == 6 bytes
+    // iclass == 8 bytes
+    // default to 6 bytes.
+    if (keylen != 4 && keylen != 6 && keylen != 8) {
+        keylen = 6;
+    }
+
+    // double up since its chars
+    keylen <<= 1;
+
+    char line[255];
+
+    // allocate some space for the dictionary
+    *data = (uint8_t*) malloc(keylen * allocation_size * sizeof(uint8_t));
```

Comment:
use  `calloc`  instead.

Reply:
Is there something like realloc in calloc?
Otherwise this would be useless.

Reply:
Or should I memset everthing new to 0 after the realloc?

Reply:
malloc vs calloc,    calloc memsets to zero.
let realloc be for now,  its for later to make sure the memory reallocated is zeroed before using.

Reply:
I'm not sure I understood your comment @iceman.
malloc + memset = calloc and realloc ~ malloc, so I'd say use:
* calloc then
* realloc + memset

Reply:
Sounds good